lib/matplotlib/axes/_axes.py

```diff
@@ -5168,9 +5176,20 @@ def _pcolorargs(funcname, *args, **kw):
         # is False.
 
         allmatch = kw.pop("allmatch", False)
+        norm = kw.pop("norm", None)
+        cmap = kw.pop("cmap", None)
+        alpha = kw.pop("alpha", 1)
 
         if len(args) == 1:
             C = np.asanyarray(args[0])
+            isBivari = isinstance(norm, (mcolors.BivariateNorm,
+                                    mcolors.BivariateColormap))
+            if (C.ndim == 3 and isBivari):
+                if cmap is None:
+                    cmap = mcolors.BivariateColormap()
+                if norm is None:
+                    norm = mcolors.BivariateNorm()
+                C = norm(C)
```

Comment:
Why do this up-front?  defering all of this to `_make_image` means the colormap and norm and be updated.